### ISSUE TITLE
[Suggestion] Download all entities with their nbt tags

### ISSUE THREAD

**Issue by DarkReaper231** [mircokroon/minecraft-world-downloader#498]
Would be useful on servers like Origin Realms for example so you could download all the npcs, armor stands with all their nbt tada, itemframes with all the paper items that have CustomModelData tags etc. I would exclude player entities tho so it doesn't download all players. Problem is that some npcs are also technically players so there would have to be some way to distinguish player from npc.

**Comment by oliviathevampire** [comment]
@mircokroon any chance? Would be so nice!

**Comment by mircokroon** [comment]
This works on some versions, but every entity type needs specific code that often breaks every game update so it is quite a bit of work to maintain. Sadly not as straightforward as "just store the NBT data".

**Comment by oliviathevampire** [comment]
> This works on some versions, but every entity type needs specific code that often breaks every game update so it is quite a bit of work to maintain. Sadly not as straightforward as "just store the NBT data".

that is true, my main problem is for example item frames not having their items in them, and armor stands not having correct rotations, mainly on 1.20.x

**Comment by Brian151** [comment]
definitely the item frames!

but yeh, understandable how their constantly making breaking changes makes things difficult
you can't even do resource packs these days without "btw, we changed this!" and wondering why your pack quit working!
